packages/libs/react-ui/src/components/Button/Button.tsx

```diff
@@ -1,87 +1,150 @@
 
-  const isLoadingAriaLiveLabel = `${
-    typeof props.children === 'string'
-      ? props.children
-      : buttonProps['aria-label'] ?? 'is'
-  } loading`.trim();
+    const iconOnly = Boolean(
+      // check if children is a ReactElement
+      (typeof children !== 'string' && typeof children !== 'number') ||
+        // check if no visuals are provided
+        (!startVisual && !endVisual && !children) ||
+        // check if only one visual is provided
+        (!children &&
+          ((startVisual && !endVisual) || (endVisual && !startVisual))) ||
+        // check if there is a loading label while loading
+        (!loadingLabel && props.isLoading),
+    );
 
-  return (
-    <button
-      {...mergeProps(buttonProps, hoverProps, focusProps)}
-      ref={ref}
-      className={classNames(
-        button({
-          variant: props.variant,
-          color: props.color,
-          isCompact: props.isCompact,
-          isLoading: props.isLoading,
-        }),
-        props.className,
-      )}
-      style={props.style}
-      title={props.title}
-      aria-disabled={props.isLoading || undefined}
-      data-disabled={props.isDisabled || undefined}
-      data-pressed={isPressed || undefined}
-      data-hovered={isHovered || undefined}
-      data-focused={isFocused || undefined}
-      data-focus-visible={isFocusVisible || undefined}
-    >
-      {props.isLoading ? (
-        <>
-          {onlyIcon ? null : 'Loading'}
-          <ProgressCircle
-            aria-hidden="true"
-            aria-label={isLoadingAriaLiveLabel}
-            isIndeterminate
-          />
-        </>
-      ) : (
-        content
-      )}
-    </button>
-  );
-};
+    const isLoadingAriaLiveLabel = `${
+      typeof children === 'string' ? children : props['aria-label'] ?? 'is'
+    } loading`.trim();
 
-export const Button = forwardRef(BaseButton);
+    return (
+      <button
+        {...mergeProps(buttonProps, hoverProps, focusProps)}
+        className={classNames(
+          button({
+            variant,
+            isCompact,
+            isLoading: props.isLoading,
+          }),
+          className,
+        )}
+        style={props.style}
+        title={props.title}
+        aria-disabled={props.isLoading || undefined}
+        data-disabled={props.isDisabled || undefined}
+        data-pressed={isPressed || undefined}
+        data-hovered={(!isPressed && isHovered) || undefined}
+        data-focused={isFocused || undefined}
+        data-focus-visible={isFocusVisible || undefined}
+        ref={ref}
+      >
+        <span
+          className={classNames({
+            [centerContentWrapper]: !iconOnly,
+            [iconOnlyStyle]: iconOnly,
+            [noEndVisualStyle]: !endVisual && !iconOnly,
+            [noStartVisualStyle]: !startVisual && !iconOnly,
+            [startVisualStyle]: startVisual && !iconOnly,
+            [endVisualStyle]: endVisual && !iconOnly,
```

Comment:
This feels like a group of styles where some are mutually exclusive. Could this be simplified?